api/app/Models/Article.php

```diff
@@ -0,0 +1,84 @@
+<?php namespace App\Models;
+
+use Spira\Repository\Collection\Collection;
+
+/**
+ *
+ * @property ArticlePermalink $permalinkRelation
+ * @property ArticlePermalink[]|Collection $previousPermalinksRelations
+ *
+ * Class Article
+ * @package App\Models
+ *
+ */
+class Article extends BaseModel
+{
+    /**
+     * The database table used by the model.
+     *
+     * @var string
+     */
+    public $table = 'articles';
+    /**
+     * The attributes that are mass assignable.
+     *
+     * @var array
+     */
+    protected $fillable = ['article_id', 'title', 'content', 'permalink', 'first_published'];
+
+    protected $hidden = ['permalinkRelation','previousPermalinksRelations'];
+
+    protected $primaryKey = 'article_id';
+
+    protected $with = ['permalinkRelation'];
+
+    protected $casts = [
+        'first_published' => 'dateTime',
+    ];
+
+    /**
+     * @param string $permalink
+     */
+    public function setPermalink($permalink)
+    {
+        $currentLink = $this->permalinkRelation;
+        if ($currentLink) {
+            $currentLink->current = false;
```

Comment:
Why do we need a permalink->current property? It is implied by it being a foreign key in the articles table. Having the current property means there is a chance of the `current` permalink not actually being the one that is the property of the model.